### ISSUE TITLE
Set a Data Point Using the New Interactive Mode

### ISSUE THREAD

**Issue by RoyiAvital** [jheinen/GR.jl#515]
Looking at the blog post [How to achieve better interactivity with GR](https://gr-framework.org/blog/posts/grm.html) I can see the data point:

![image](https://user-images.githubusercontent.com/7347975/235285166-9d97e5d8-2a32-4b46-8d4b-490e2e536ec8.png)

I was wondering if there could be a feature to set them.
Namely I want some points to be displayed constantly.

For instance, on MATLAB I can do the following (From [Orthogonal Projection onto the Unit Simplex](https://math.stackexchange.com/questions/2402504)):

![image](https://user-images.githubusercontent.com/7347975/235285231-f5de13af-78be-4a49-999c-31e4a3979a65.png)

On MATALB I can hit Ctrl and make the one displayed permanent and move forward to a new one.

**Comment by jheinen** [comment]
The edit feature is still WIP.

**Comment by IngoMeyer441** [comment]
I think what we could implement next week is to show multiple data points at once. This should only require a few changes.

**Comment by RoyiAvital** [comment]
By the way, will those data point work on images as well?

**Comment by IngoMeyer441** [comment]
> By the way, will those data point work on images as well?

Yes, they work on these GR plot types:

- `line`
- `scatter`
- `stem`
- `step`
- `heatmap`
- `marginalheatmap`
- `contour`
- `contourf`
- `imshow`
- `pie`
- `hexbin`
- `shade`
- `quiver`

> On MATALB I can hit Ctrl and make the one displayed permanent and move forward to a new one.

Do you really use Ctrl? [Regarding the MATLAB documentation](https://de.mathworks.com/help/matlab/ref/matlab.graphics.shape.internal.datacursormanager.html), the Shift key can be used for this:

> Click a plotted data point. To create multiple data tips, hold the Shift key as you select data points. To move a data tip to another data point, drag the point where the data tip is located.

**Comment by IngoMeyer441** [comment]
Starting with release [v0.72.5](https://github.com/jheinen/GR.jl/releases/tag/v0.72.5), multiple tooltips can be shown when holding the Alt key (option key on macOS). Pressing the shift key shows all y values for the current x value in a combined tooltip.

**Comment by RoyiAvital** [comment]
Is it available using `Plots.jl` or only directly using `GR.jl`?

I think you fixed one of the most important issues in the plotting eco system.
I really missed this feature from MATLAB.

**Comment by IngoMeyer441** [comment]
This is only available in GR.jl with the plots display:

```julia
ENV["GRDISPLAY"] = "plot"
using GR
```

**Comment by RoyiAvital** [comment]
@IngoMeyer441 , I am trying the new plot display on Windows.
Yet nothing pops. It opens a new CMD window yet nothing happens.

**Comment by IngoMeyer441** [comment]
@RoyiAvital Could you describe, which steps you executed? Hopefully, I can reproduce your problem.

> Yet nothing pops. It opens a new CMD window yet nothing happens.

So you don't get any plotting window or is this related to non-existing tooltips?

**Comment by RoyiAvital** [comment]
Nothing shows up at all.

**Comment by IngoMeyer441** [comment]
![image](https://github.com/jheinen/GR.jl/assets/1449087/777816f5-ae81-4f93-b53e-482dabe5c13e)

Ok, I get the same broken behavior on Windows 11 with Julia 1.9.2 and latest GR.jl :disappointed:

**Comment by IngoMeyer441** [comment]
On a second try, I get a plot window:

![image](https://github.com/jheinen/GR.jl/assets/1449087/007c290a-ca04-4f1f-a8a5-deef29fb13ba)


When starting grplot for the first time, the Windows firewall blocked the internal TCP traffic and asked for permissions. This probably caused a timing problem. Did the Windows firewall dialog also show up for you @RoyiAvital?

**Comment by RoyiAvital** [comment]
OK,
Now I have something strange.

I closed Julia and started new session:

```julia
ENV["GRDISPLAY"] = "plot"
using GR;
plot(randn(20,5), labels=("a", "b", "c", "d", "e"))
```

A CMD window appeared.
After a second a plot window appeared.
I could see the new labels when walking on the data (With `Shift` I could see all data points at the given `x`).

I closed the plot window. The CMD closed as well.
I entered `plot(randn(20,5), labels=("a", "b", "c", "d", "e"))` again.
This time nothing happens. Also for trying again and again,

So the first plot in a session works.
But nothing after that.

**Comment by IngoMeyer441** [comment]
@jheinen AFAIK, it is intended behavior, that new plots after closing grplot do not open a new window, isn't it?

**Comment by RoyiAvital** [comment]
@IngoMeyer441 , Why?
Also, Does it mean only single figure can be seen at a time?

**Comment by IngoMeyer441** [comment]
Yes, currently, a new figure overwrites an older figure in the grplot window. Maybe, we should discuss, if this is a good default behavior or if it should at least be possible to change this behavior (for example with an environment variable).

**Comment by jheinen** [comment]
One could imagine leaving "frozen" versions of figures on the screen - but without interaction. With that you could at least compare figures.

To support multiple plots with full interaction would also require extensive changes in `Plots.jl`.

**Comment by RoyiAvital** [comment]
@jheinen , What about multiple plots with interaction with `GRUtils.jl`?

**Comment by jheinen** [comment]
We will discuss that in our group. If full interaction is desired, however, this would be a very extensive expansion that we cannot implement in the short term.
